src/utlist.h

```diff
@@ -349,6 +349,26 @@ do {
   }                                                                                            \
 } while (0)
 
+#define LL_INSERT_INORDER(head,add,cmp)                                                        \
+  do {                                                                                         \
+    LDECLTYPE(head) _pos;                                                                      \
+    _CASTASGN(_pos, head);                                                                     \
+    LL_FIND_POS(_pos,add,cmp);                                                                 \
```

Comment:
I think I'd like to see the phase `FIND_POS` replaced throughout with `LOWER_BOUND`, as in [std::lower_bound](http://en.cppreference.com/w/cpp/algorithm/lower_bound). That will help people remember what exactly it does. (E.g. it does *not* return NULL if the element isn't found; it *does* return specifically the first instance of the element if there are multiple consecutive copies of it; etc.)